app/uk/gov/hmrc/slacknotifications/services/NotificationService.scala

```diff
@@ -106,15 +107,33 @@ class NotificationService @Inject()(
 
   private[services] def sendSlackMessage(slackMessage: SlackMessage)(
     implicit hc: HeaderCarrier): Future[NotificationResult] =
-    slackConnector.sendMessage(slackMessage).map { response =>
-      response.status match {
-        case 200 => NotificationResult().addSuccessfullySent(slackMessage.channel)
-        case _ =>
-          val slackError = SlackError(response.status, response.body)
-          Logger.warn(slackError.message)
-          NotificationResult().addError(slackError)
+    slackConnector
+      .sendMessage(slackMessage)
+      .map { response =>
+        response.status match {
+          case 200 => NotificationResult().addSuccessfullySent(slackMessage.channel)
+          case _   => logAndReturnSlackError(response.status, response.body, slackMessage.channel)
+        }
       }
-    }
+      .recover(handleSlackExceptions(slackMessage.channel))
+      .recoverWith {
+        case NonFatal(ex) =>
+          Logger.error(s"Unable to notify Slack channel ${slackMessage.channel}", ex)
+          throw ex
+      }
+
+  private def handleSlackExceptions(channel: String): PartialFunction[Throwable, NotificationResult] = {
+    case ex: BadRequestException => logAndReturnSlackError(400, ex.message, channel)
+    case ex: Upstream4xxResponse => logAndReturnSlackError(ex.upstreamResponseCode, ex.message, channel)
+    case ex: NotFoundException   => logAndReturnSlackError(404, ex.message, channel)
+    case ex: Upstream5xxResponse => logAndReturnSlackError(ex.upstreamResponseCode, ex.message, channel)
+  }
```

Comment:
Error handling of pure slack integration could be done inside of `SlackConnector` but it's also fine here